blog/_posts/2021-06-17-uncertainty.md

```diff
@@ -0,0 +1,244 @@
+---
+layout: review
+title: "Uncertainty Estimation Review"
+tags: machine-learning essentials blog
+author: Thierry Judge
+---
+
+# Introduction
+
+This post is a review of uncertainty estimation methods with an emphasis on classification and segmentation. 
+
+# Evaluating Uncertainty
+
+## Model calibration
+
+A neural network trained for classification is said to be calibrated if its confidence is equal to the probability of 
+it being correct. For example, given 100 predictions with an average confidence of 0.7, 70 predictions should be correct. 
+This can be expressed :
+
+$$
+\mathbb{P} ( \hat{Y} = Y | \hat{P} = P ) = p  \text{   } \forall p \in [0, 1]
+$$
+
+where $$\hat{Y}$$ is the predicted class and $$\hat{P}$$ is the predicted probability.
+
+Calibration is often shown with reliability diagrams [^5] (see figures in the following sections). These diagrams are built by 
+splitting the samples into bins according to their confidence. The accuracy for each bin each calculated and should 
+correspond to the average confidence. The difference between the accuracy and confidence is known as the gab. Averaging
+the gab for all bins gives the expected calibration error (ECE) [^6]. 
+
+## Uncertainty-error overlap
+In the case of segmentation, the uncertainty of a prediction can be predicted via an uncertainty map. An uncertainty map is an
+uncertainty prediction for each pixel. Uncertainty map quality can be evaluated by computing the uncertainty-error 
+overlap. Uncertainty-error overlap is computed by obtaining the dice score of the uncertainty map and a pixel-wise error
+map between the prediction and the groundtruth segmentation map. 
```

Comment:
A picture would help here.

Reply:
Is the picture added in eacdc9f clear ?

Reply:
Not sure if the thresholded uncertainty added in eacdc9f shows the uncertainty-error overlap, but anyways the description is clear having re-read it.